### PR TITLE
[MINOR] Remove ScalarFunction from datafusion.proto #10173

### LINKED ISSUE #10173: Remove ScalarFunction from datafusion.proto
### Is your feature request related to a problem or challenge?

ScalarFunction no longer has any use in the proto definition and can be removed. Noted in https://github.com/apache/datafusion/pull/10098#discussion_r1573857862

### Describe the solution you'd like

Proto file and code references updated to remove ScalarFunction.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dmitrybugakov**
## Which issue does this PR close?

Closes #10173

### CODE REVIEW

**Comment by alamb** [comment]
Thank you @dmitrybugakov

**Review by alamb — APPROVED** [review]
A very nice first PR indeed ❤️ Thanks again!

Screenshot 2024-04-23 at 1 40 09 PM

**Comment by Omega359** [comment]
lgtm

**Comment by alamb** [comment]
Thanks again @dmitrybugakov

File: datafusion/proto/proto/datafusion.proto [hunk 1]
```diff
@@ -389,7 +389,6 @@ message LogicalExprNode {
     NegativeNode negative = 13;
     InListNode in_list = 14;
     Wildcard wildcard = 15;
-    ScalarFunctionNode scalar_function = 16;
```
Comment (liukun4515) [inline datafusion/proto/proto/datafusion.proto:392]: I think we should add comment to occupy this position of `16` @dmitrybugakov @alamb @Omega359

File: datafusion/proto/proto/datafusion.proto [hunk 2]
```diff
@@ -1458,7 +1310,6 @@ message PhysicalExprNode {
     PhysicalSortExprNode sort = 10;
     PhysicalNegativeNode negative = 11;
     PhysicalInListNode in_list = 12;
-    PhysicalScalarFunctionNode scalar_function = 13;
```
Comment (liukun4515) [inline datafusion/proto/proto/datafusion.proto:1461]: same comments

**Pull request merged**